### PR TITLE
Refactor Group model

### PR DESCRIPTION
**Pull request by vpereira**
Error Message if user not found:

Screenshot_2020-11-04_10-27-21

Existing user:

Screenshot_2020-11-04_10-27-52

### CODE REVIEW

File: src/api/app/models/group.rb [hunk 1]
```diff
@@ -34,7 +32,9 @@ class Group < ApplicationRecord
   alias_attribute :name, :title
 
   def self.find_by_title!(title)
-    find_by_title(title) || (raise NotFoundError, "Couldn't find Group '#{title}'")
+    find_by!(title: title)
+  rescue ActiveRecord::RecordNotFound => e
```
Comment (vpereira) [inline src/api/app/models/group.rb:36]: any better way to change the message of the raised exception?

Comment (dmarcoux) [reply src/api/app/models/group.rb:36]: I don't think so.

I've found this [StackOverflow thread](https://stackoverflow.com/questions/2823748/how-do-i-add-information-to-an-exception-message-in-ruby) which is providing more info.

File: src/api/app/models/group.rb [hunk 2]
```diff
@@ -34,7 +32,9 @@ class Group < ApplicationRecord
...
+    raise e, "Couldn't find Group '#{title}'"
```
Comment (dmarcoux) [inline src/api/app/models/group.rb:37]: To follow up on my previous comment,  we should also include the backtrace like this:

```suggestion
    raise e, "Couldn't find Group '#{title}'", e.backtrace
```

File: src/api/app/models/group.rb (hunk 1 above)
Comment (dmarcoux) [reply src/api/app/models/group.rb:36]: I don't think so.

I've found this [StackOverflow thread](https://stackoverflow.com/questions/2823748/how-do-i-add-information-to-an-exception-message-in-ruby) which is providing more info.

Comment (hennevogel) [reply src/api/app/models/group.rb:36]: I take it you checked how the API is handling `ActiveRecord::RecordNotFound` right?

Comment (vpereira) [reply src/api/app/models/group.rb:36]: @hennevogel 

```
rescue_from Backend::NotFoundError, ActiveRecord::RecordNotFound do |exception|
      render_error message: exception.message, status: 404, errorcode: 'not_found'
    end
```
Is there something missing? Any test that you want to see?

Comment (hennevogel) [reply src/api/app/models/group.rb:36]: Nope, all is good. I was just not sure.

File: src/api/app/models/group.rb [hunk 3]
```diff
@@ -80,16 +81,13 @@ def update_from_xml(xmlhash)
...
+      delete_user(GroupsUser, login_id, id)
     end
   end
 
   def add_user(user)
-    return if users.find_by_id(user.id) # avoid double creation
-
-    gu = GroupsUser.create(user: user, group: self)
-    gu.save!
+    return if users.find_by_id(user.id) || GroupsUser.create!(user: user, group: self)
```
Comment (dmarcoux) [inline src/api/app/models/group.rb:90]: Is `return if` really needed here? It could be removed unless I'm missing something.

**Review by dmarcoux — APPROVED** [review]

File: src/api/app/models/group.rb (hunk 1 above)
Comment (vpereira) [reply src/api/app/models/group.rb:36]:
```
(base) vpereira@linux-qouy:~/open-build-service> dosc api '/group/xxxx'
WARNING: SSL certificate checks disabled. Connection is insecure!

Server returned an error: HTTP Error 404: Not Found
Couldn't find Group 'xxxx'
```

File: src/api/app/models/group.rb (hunk 3 above)
Comment (vpereira) [reply src/api/app/models/group.rb]: done!

**Review by dmarcoux — APPROVED** [review]

**Pull request merged**